Sources/swift-doc/Supporting Types/Page.swift

```diff
@@ -70,11 +70,7 @@ func writeFile(_ data: Data, to url: URL) throws {
 
 fileprivate extension URLComponents {
     func appendingPathComponent(_ component: String) -> URLComponents? {
-        var urlComponents = self
-        var pathComponents = urlComponents.path.split(separator: "/").map { "\($0)" }
-        pathComponents.append(component)
-        urlComponents.path = "/" + pathComponents.joined(separator: "/")
-
-        return urlComponents
+        return url.map { $0.appendingPathComponent(component) }
+            .flatMap { URLComponents(url: $0, resolvingAgainstBaseURL: true) }
```

Comment:
I slightly prefer the solution in #139 (though I need to test it out a bit more to make sure they're equivalent). I think we can all agree though that my code here, in addition to being buggy, is totally bizarre and anything would be an improvement.

Reply:
I've been staring at the two implementations for a few minutes now. Ultimately I wish there were unit tests to prove it, but I like that #139 does away with the `URLComponents` abstraction entirely. It sure looks to me like the implementation in #139 introduces a bug where all paths become relative, though -- note that it takes the `.path` of the URL instead of the `.absoluteString`.

I think we can have that implementation if `.path` is swapped for `.absoluteString` and everything will be better all around. I proved this to myself in a Playground, at least.

```swift
let baseURLs = [
    "https://website.com",
    "./",
    "/"
]

let identifierStandin = "identifier_standin"

let mattsURLs = baseURLs.map { URLComponents(string: $0)?.url.flatMap { URLComponents(url: $0.appendingPathComponent(identifierStandin), resolvingAgainstBaseURL: true)?.string } }
let karesURLs = baseURLs.map { URL(string: $0)?.appendingPathComponent(identifierStandin).path }

let bestOfBothWorlsURLs = baseURLs.map { URL(string: $0)?.appendingPathComponent(identifierStandin).absoluteString }

func pretty(_ arr: [String?]) -> String {
    arr.map { $0 ?? "nil" }.joined(separator: "\n")
}

print(pretty(mattsURLs))
print("====")
print(pretty(karesURLs))
print("====")
print(pretty(bestOfBothWorlsURLs))
```

```
https://website.com/identifier_standin
./identifier_standin
/identifier_standin
====
/identifier_standin
./identifier_standin
/identifier_standin
====
https://website.com/identifier_standin
./identifier_standin
/identifier_standin
```

Reply:
I merged with #139 and then made the aforementioned change from `.path` to `.absoluteString`. Should make it a clean merge whether you merge #139 before merging this or not.